nemo/collections/nlp/nm/data_layers/state_tracking_sgd_datalayer.py

```diff
@@ -0,0 +1,110 @@
+        num_noncategorical_slots (int): Number of non-categorical slots present in the service
+        noncategorical_slot_status (int): The status of each non-categorical slot in the service
+        noncategorical_slot_value_start (int): The index of the starting subword corresponding to the slot span for a non-categorical slot value
+        noncategorical_slot_value_end (int): The index of the ending (inclusive) subword corresponding to the slot span for a non-categorical slot value
+        start_char_idx (int): Start character indices in the original utterance corresponding to the tokens
+        end_char_idx (int): Inclusive end character indices in the original utterance corresponding to the tokens
+        num_slots (int): Total number of slots present in the service
+        requested_slot_status (int): Takes value 1 if the corresponding slot is requested, 0 otherwise
+        num_intents (int): Total number of intents present in the service
+        intent_status_labels (int): Intent labels
+
+        """
+        return {
+            "example_id_num": NeuralType(('B'), ChannelType()),
+            "service_id": NeuralType(('B'), ChannelType()),
+            "is_real_example": NeuralType(('B'), ChannelType()),
+            "utterance_ids": NeuralType(('B', 'T'), ChannelType()),
+            "utterance_segment": NeuralType(('B', 'T'), ChannelType()),
+            "utterance_mask": NeuralType(('B', 'T'), ChannelType()),
+            "num_categorical_slots": NeuralType(('B'), LengthsType()),
+            "categorical_slot_status": NeuralType(('B', 'T'), LabelsType()),
+            "num_categorical_slot_values": NeuralType(('B', 'T'), LengthsType()),
+            "categorical_slot_values": NeuralType(('B', 'T'), LabelsType()),
+            "num_noncategorical_slots": NeuralType(('B'), LengthsType()),
+            "noncategorical_slot_status": NeuralType(('B', 'T'), LabelsType()),
+            "noncategorical_slot_value_start": NeuralType(('B', 'T'), LabelsType()),
+            "noncategorical_slot_value_end": NeuralType(('B', 'T'), LabelsType()),
+            "start_char_idx": NeuralType(('B', 'T'), LabelsType()),
+            "end_char_idx": NeuralType(('B', 'T'), LabelsType()),
+            "num_slots": NeuralType(('B'), LengthsType()),
+            "requested_slot_status": NeuralType(('B', 'T'), LabelsType()),
+            "num_intents": NeuralType(('B'), LengthsType()),
+            "intent_status_labels": NeuralType(('B'), LabelsType()),
+        }
+
+    def __init__(
+        self,
+        dataset_split,
+        dialogues_processor,
+        dataset_type=SGDDataset,
+        shuffle=False,
+        batch_size=1,
+        num_workers=-1,
+        pin_memory=False,
+    ):
+
+        dataset_params = {
+            'dataset_split': dataset_split,
+            'dialogues_processor': dialogues_processor,
+        }
+        super().__init__(
+            dataset_type,
+            dataset_params,
```

Comment:
how are you making sure that "dataset_type" injected here will return batch of samples having the same format as the one described in output ports of that DL?

@okuchaiev any comments on that pattern?